packages/@aws-cdk/aws-ecs/lib/images/asset-image.ts

```diff
@@ -24,6 +26,10 @@ export class AssetImage extends ContainerImage {
     this.asset.repository.grantPull(containerDefinition.taskDefinition.obtainExecutionRole());
   }
 
+  public renderRepositoryCredentials(): undefined {
```

Comment:
jsii should give you an error here because you changed the return type of this function while overriding it. C# does not allow narrowing the return type, so we have to disallow that everywhere as well.

You need to declare the return type as `CfnTaskDefinition.RepositoryCredentialsProperty | undefined`, even though we could give a narrower type.